docs/user-manual/engine/migrations.md

```diff
@@ -0,0 +1,77 @@
+# Engine Migrations
+
+The PlayCanvas Engine is continually evolving, and its updates occasionally introduce breaking changes that require users to adapt their scripts.
+
+This guide provides an overview of all major breaking changes across releases, offering users a helpful resource for migrating their code to newer versions.
+
+## Migration from 1.75.0 to 2.4.0
+
+The migration from major version 1 to major version 2 is a substantial update, introducing numerous breaking changes.
+
+### Breaking changes
+
+#### WebGL 1
+
+Support for WebGL1 has been discontinued; the engine now exclusively supports WebGL2 and WebGPU (in beta). If your application relies on WebGL1, you will need to continue using version 1 of the engine.
+
+#### AudioSourceComponent
+
+The AudioSourceComponent, which was replaced by the SoundComponent some time ago, has now been completely removed.
+
+#### Legacy Scripts
+
+The legacy scripting system, deprecated since 2016 and maintained in a read-only state for several years, is now being removed entirely.
+
+#### Deprecated functions
+
+Numerous deprecated functions that provided backward compatibility have been removed. If your application displays deprecation warnings when using the debug version of Engine 1, these issues should be resolved before migrating to Engine 2. In Engine 2, deprecated warnings are no longer displayed, and the backward compatibility code has been eliminated.
+
+#### Basic Material
+
+The BasicMaterial has been removed. To achieve equivalent functionality, you can use a StandardMaterial with the emissive color or emissive map set as a replacement.
+
+#### Rendering to a texture
+
+When rendering to a texture, the deprecated method of configuring the RenderTarget on the Layer has been fully removed. The RenderTarget must now be set directly on the camera instead.
+
+#### Engine rendering callbacks
+
+The engine previously executed multiple callbacks per frame for camera and layer rendering. These have been replaced with an event-driven system that supports multiple subscribers. The new events are now emitted by the Scene class. For more details, refer to [this pull request](https://github.com/playcanvas/engine/pull/7156).
+
+#### StandardMaterial tint flags
+
+The tint options for StandardMaterial have been confusing and inconsistent, so we've removed the flags for Ambient, Diffuse, Specular and Emissive tint. Previously, these flags only affected certain cases, such as when a texture was applied. With this update, tint colors are now always applied. To disable tinting, set the color to a neutral value (the default tint colors used when creating a new material):
```

Comment:
I thought we had kept `specularTint`.

Reply:
The last sentence isn't quite true with respect to `emissiveTint`. By default, it's black. But to disable tinting, you must set it to white.